cmd/push.go

```diff
@@ -0,0 +1,97 @@
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	"github.com/flanksource/canary-checker/pkg"
+	"github.com/flanksource/canary-checker/pkg/cache"
+	"github.com/flanksource/canary-checker/pkg/push"
+	"github.com/flanksource/commons/logger"
+	"github.com/spf13/cobra"
+)
+
+var name, description, checkType, status, duration, message, checkDetails, errorMessage string
+var Push = &cobra.Command{
+	Use:   "push",
+	Short: "Push a check to the canary-checker-server",
+	Run: func(cmd *cobra.Command, servers []string) {
+		var passed bool
+		var details []byte
+		if strings.ToLower(status) == "passed" {
+			passed = true
+		}
+		checkDuration, err := time.ParseDuration(duration)
+		if err != nil {
+			logger.Errorf("Invalid duration: %s", err)
+		}
+		if checkDetails != "" {
+			detailsContent, err := ioutil.ReadFile(checkDetails)
+			if err != nil {
+				logger.Errorf("Failed to read check details: %s", err)
+			} else {
+				details, err = json.Marshal(detailsContent)
+				if err != nil {
+					logger.Errorf("Failed to marshal check details: %s", err)
+				}
+			}
+
+		}
+		for _, server := range servers {
+			checkStatus := pkg.CheckStatus{
+				Status:   passed,
+				Duration: int(checkDuration.Milliseconds()),
+				Time:     time.Now().UTC().Format(time.RFC3339),
+				Message:  message,
+				Detail:   details,
+			}
+			check := pkg.Check{
+				Name:        name,
+				Type:        checkType,
+				Key:         fmt.Sprintf("%s/%s/%s", checkType, name, description),
+				Description: description,
+				ID:          fmt.Sprintf("%s/%s/%s", checkType, name, description),
+			}
+			if c := cache.CacheChain.GetCheckFromKey(check.Key); c != nil {
+				if passed {
+					check.Uptime.Passed += 1
+				} else {
+					check.Uptime.Failed += 1
+				}
+			} else {
+				if passed {
+					check.Uptime.Passed = 1
+				} else {
+					check.Uptime.Failed = 1
+				}
+			}
```

Comment:
I don't see how this will ever be populated ?

Reply:
Yeah it would never be executed